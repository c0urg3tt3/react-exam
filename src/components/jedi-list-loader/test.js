import React from 'react'

import { shallow } from 'enzyme'

import JediListLoader, { loaderQuoteProps } from './component'

describe('[component] JediListLoader', () => {
  it('should render component', () => {
    const wrapper = shallow(
      <JediListLoader>
      { ({message, quote, autor}) => {
        return (
          <div>
            <p className="message">{message}</p>
            <p className="quote">{quote}</p>
            <p className="autor">{autor}</p>
          </div>
        )
      }}
      </JediListLoader>
    )

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('div')).toEqual(true)
    expect(wrapper.hasClass('jedi-list-loader')).toEqual(true)

    expect(wrapper.find('p')).toHaveLength(3)

    expect(wrapper.find('.message').text()).toEqual(loaderQuoteProps.message)
    expect(wrapper.find('.quote').text()).toEqual(loaderQuoteProps.quote)
    expect(wrapper.find('.autor').first().text()).toEqual(loaderQuoteProps.autor)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
