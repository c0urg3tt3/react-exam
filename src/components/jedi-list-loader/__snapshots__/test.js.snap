// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] JediListLoader should render component 1`] = `"<div class=\\"jedi-list-loader\\"><div><p class=\\"message\\">C3PO is fetching jedi list ...</p><p class=\\"quote\\">hnnnhrrhhh huurh uughghhhgh raaaaaahhgh uughguughhhghghghhhgh</p><p class=\\"autor\\">Chewbacca</p></div></div>"`;
